setup/advanced/kubernetes-sidecar.md

```diff
@@ -0,0 +1,119 @@
+# Kubernetes Sidecar Container Setup
+
+To get started with a Kubernetes sidecar container setup, simply adapt the following deployment yaml to suit your needs. Be sure to create a persistent volume and persistent volume claim for Gluetun to be able to store data!
+
+```yaml
+---
+ apiVersion: apps/v1
+ kind: Deployment
+ metadata:
+   name: gluetun-deployment
+   labels:
+     app: gluetun
+ spec:
+   replicas: 1
+   strategy:
+       type: RollingUpdate
+       rollingUpdate:
+          maxSurge: 1
+   selector:
+      matchLabels:
+        app: gluetun
+   template:
+     metadata:
+       labels:
+         app: gluetun
+     spec:
+      containers:
+         - name: gluetun
+           image: ghcr.io/qdm12/gluetun # Optionally you can use the "qmcgaw/gluetun" image as well as specify what version of Gluetun you desire
+           imagePullPolicy: Always
+           securityContext:
+             capabilities:
+               add: ["NET_ADMIN"]
+           # You can optionally specify ports that are used by your containers, but it is not strictly necessary
+           # ports:
+             # - containerPort: 8080
+
+           # Adapt the following environment variables to suit your needs and VPN provider
+           env:
```

Comment:
Don't you need `DNS_KEEP_NAMESERVER=on` for the gluetun sidecar to work?

There was a rather big fuss about it in https://github.com/qdm12/gluetun/issues/1523 🤔 

I personally don't use gluetun with Kubernetes so I can't really tell 😢

Reply:
`DNS_KEEP_NAMESERVER=on ` is absolutely needed if the container is expected to be able to resolve or communicate to any cluster services, without it cluster services cannot be resolved consistently or at all in many setups.